src/porepy/utils/interpolation_tables.py

```diff
@@ -345,32 +428,351 @@ def diff(self, x, axis):
+
+            # Find which rows (dimensions) were impacted by rounding errors
+            rows_with_repeats = np.where(
+                np.any(rounding_error_danger[:, columns], axis=1)
+            )[0]
+
+            # Only add indices if there were points that could have been rounded
+            # wrongly.
+            if np.any(rows_with_repeats):
+                # Storage of extra indices.
+                extra_ind = []
+
+                # Loop over all combinations of dimensions that could have been
+                # afflicted by rounding errors.
+                # Referring to the example at the begining of this method, if (5, 5) was
+                # rounded to (4, 4), we need to add 1 to (4, 4) on both the first and
+                # second axis, as well as their combination. If instead (5, 5) became
+                # (5, 4), we only add on the second axis.
+                # We achieve this by invoking itertools.combinations on correctly
+                # sized arrays.
+                for combination_length in range(1, rows_with_repeats.size + 1):
+                    for active_rows in itertools.combinations(
+                        rows_with_repeats, combination_length
+                    ):
+                        # Find which columns should be fixed for this combination of
+                        # rows.
+                        # The list is needed since itertools.combinations returns a
+                        # tuple that is not fit for array indexing.
+                        active_columns = np.all(
+                            np.atleast_2d(rounding_error_danger[list(active_rows)]),
+                            axis=0,
+                        )
+
+                        # Do a copy (!) and increase the index.
+                        tmp_ind = full_ind[:, active_columns].copy()
+                        tmp_ind[list(active_rows)] += 1
+                        extra_ind.append(tmp_ind)
+
+                # Collect data, prepare for return.
+                full_ind = np.hstack((full_ind, np.hstack(extra_ind)))
+
+        return full_ind
+
+    def __repr__(self) -> str:
+        """String representation"""
+        s = f"Adaptive interpolation table in {self._param_dim} dimensions.\n"
+        s += f"The table stores function values in {self._pt.shape[1]} points.\n"
+        s += "The minimum coordinates in each dimension (possibly combining multiple "
+        s += "coordinates) are: \n \n \t"
+        for dim in range(self._param_dim):
+            s += f"{dim}: {self._pt[dim].min()}, "
+        s = s[:-2] + "\n \n"
+        s += "The maximum coordinates in each dimension (possibly combining multiple "
+        s += "coordinates) are: \n \n \t"
+        for dim in range(self._param_dim):
+            s += f"{dim}: {self._pt[dim].max()}, "
+        s = s[:-2] + "\n \n"
+
+        s += f"Minimum function value: {np.min(self._values)}\n"
+        s += f"Maximum function value: {np.max(self._values)}"
```

Comment:
```suggestion
        if self._pt.size != 0 and self._values.size != 0:
            s += "The minimum coordinates in each dimension (possibly combining multiple "
            s += "coordinates) are: \n \n \t"
            for dim in range(self._param_dim):
                s += f"{dim}: {self._pt[dim].min()}, "
            s = s[:-2] + "\n \n"
            s += "The maximum coordinates in each dimension (possibly combining multiple "
            s += "coordinates) are: \n \n \t"
            for dim in range(self._param_dim):
                s += f"{dim}: {self._pt[dim].max()}, "
            s = s[:-2] + "\n \n"
            s += f"Minimum function value: {np.min(self._values)}\n"
            s += f"Maximum function value: {np.max(self._values)}"
```
In some cases, after instantiation of `AdaptiveInterpolationTable` `self._pt` and `self._values` are empty.